core/src/main/java/hudson/model/Run.java

```diff
@@ -332,7 +332,7 @@ public void reload() throws IOException {
      */
     @SuppressWarnings("deprecation")
     protected void onLoad() {
-        for (Action a : getActions()) {
+        for (Action a : getAllActions()) {
```

Comment:
this seems very wrong.  `getAllActions()` will involve actions from the `TransientActionFactory` whose actions should never be persisted so calling `onLoad` here seems wrong by design (anyone depending on this has a questionable `TransientActionFactory`/`Action` implementation)?
@jglick?

Reply:
Probably unnecessary—typically you would do any initialization in the factory rather than marking it as a `RunAction2`—but should also be harmless.

Reply:
Thanks for the clarification.